### PR TITLE
[1] - Generic Error in getCoin

### LINKED ISSUE #1: [UH: 1] /coin/status/{coin_id}
Desc: Get given coin status
Expected Result:
[
   {
     "coin_id":  "string",
     "symbol": "string",
     "name": "string",
      "name_id": "string",
       "rank": 0,
      "price_usd": 0,
   }
]

### PR DESCRIPTION
**Pull request by crackergirl**
I don't know how to check the value of the cryptocurrency because it changes over time.

### CODE REVIEW

File: tests/app/Infrastructure/Controller/CryptoDataControllerTest.php [hunk 1]
```diff
@@ -3,11 +3,32 @@
...
 
+use App\Application\CoinLoreCryptoDataSource\CoinLoreCryptoDataSource;
+use Illuminate\Http\Response;
 use Tests\TestCase;
-
+use Exception;
+use Mockery;
 
 class CryptoDataControllerTest extends TestCase
 {
```
Comment (ivans1516) [inline tests/app/Infrastructure/Controller/CryptoDataControllerTest.php]: 2 test are missing

File: tests/app/Infrastructure/Controller/CryptoDataControllerTest.php [hunk 2]
```diff
@@ -3,11 +3,32 @@
...
 
+    /**
+     * @test
+     */
+    public function genericError()
+    {
+        $this->coinLoreCryptoDataSource = Mockery::mock(CoinLoreCryptoDataSource::class);
```
Comment (ivans1516) [inline tests/app/Infrastructure/Controller/CryptoDataControllerTest.php]: add setup

File: app/Infrastructure/Controllers/CryptoDataController.php [hunk 3]
```diff
@@ -17,20 +18,18 @@ public function __construct(CoinLoreService $coinLoreService)
...
+    public function __invoke(string $id_coin): JsonResponse
     {
         try {
-            $coin = $this->coinLoreService->execute($coin);
+            $coin = $this->coinLoreService->execute($id_coin);
 
         }catch (Exception $exception) {
             return response()->json([
                 'error' => $exception->getMessage()
             ], Response::HTTP_BAD_REQUEST);
```
Comment (ivans1516) [inline app/Infrastructure/Controllers/CryptoDataController.php]: GetCode function is better

File: app/Domain/Coin.php [hunk 4]
```diff
@@ -45,9 +46,14 @@ public function getValueUsd(): float
         return $this->value_usd;
     }
 
-
-    function jsonSerialize()
+    public function toJson(): string|false
     {
-        return get_object_vars($this);
```
Comment (jalex3421) [inline app/Domain/Coin.php]: what a smooth move mr Ivan!!

File: app/Infrastructure/Controllers/CryptoDataController.php [hunk 5]
```diff
@@ -26,10 +26,9 @@ public function __invoke(string $id_coin): JsonResponse
         }catch (Exception $exception) {
             return response()->json([
```
Comment (jalex3421) [inline app/Infrastructure/Controllers/CryptoDataController.php]: The name of the class should change

File: tests/app/Application/CoinLoreAPI/CoinLoreServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<?php
+
```
Comment (jalex3421) [inline tests/app/Application/CoinLoreAPI/CoinLoreServiceTest.php]: Class name should change as soon as possible!

File: tests/app/Infrastructure/Controller/CryptoDataControllerTest.php [hunk 7]
```diff
@@ -7,28 +7,74 @@
 use Illuminate\Http\Response;
 use Tests\TestCase;
```
Comment (jalex3421) [inline tests/app/Infrastructure/Controller/CryptoDataControllerTest.php]: Class name should change ASAP

File: app/Domain/Coin.php [hunk 8]
```diff
@@ -2,8 +2,9 @@
 
 namespace App\Domain;
 use JsonSerializable;
+use PHPUnit\Util\Json;
 
-class Coin implements JsonSerializable
+class Coin
 {
     private string $coin_id;
     private string  $name;
```
Comment (jalex3421) [inline app/Domain/Coin.php:10]: Also change the parameters Mr Ivan. Hope you kill this little refactor :)

**Review by jalex3421 — COMMENTED** [review]
Good job team!

**Pull request merged**